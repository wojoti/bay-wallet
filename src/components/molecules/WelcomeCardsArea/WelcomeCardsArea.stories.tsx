import {ComponentMeta, ComponentStory} from '@storybook/react-native';

import WelcomeCardsArea from './WelcomeCardsArea';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'molecules/WelcomeCardsArea',
  component: WelcomeCardsArea,
  argTypes: {
    onButtonPress: {action: 'onButtonPress'},
  },
} as ComponentMeta<typeof WelcomeCardsArea>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof WelcomeCardsArea> = args => (
  <WelcomeCardsArea {...args} />
);

export const Example = Template.bind({});
Example.args = {
  mainText: 'New cards in few clicks',
  cardText:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.',
  testId: '',
};
