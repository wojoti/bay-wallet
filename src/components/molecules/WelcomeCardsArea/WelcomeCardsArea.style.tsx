import {StyleSheet} from 'react-native';

const style = StyleSheet.create({
  wrapper: {
    width: '100%',
    alignItems: 'center',
  },
  cardSizer: {width: '85%'},
  topSpacer: {
    marginTop: 50,
  },
  labelSpacer: {
    marginTop: 20,
  },
  buttonSpacer: {
    marginTop: 30,
  },
  headerSizer: {
    width: '78%',
  },
  logoSizer: {
    width: 'auto',
  },
  btnStyle: {
    borderWidth: 0,
  },
});
export default style;
