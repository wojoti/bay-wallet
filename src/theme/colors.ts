export const lightPrimary = '#4E439E';
export const primary = '#363062';
export const secondary = 'white';

export const darkGrey = '#595959';
export const grey = '#6B6B6B';
export const vlightGrey = '#D1D1D1';
export const borderGrey = '#E2E2E2';

export const iconGrey = '#BDBDBD';

export const searchGrey = '#F5F5F5';

export const separator = '#F6F6F6';

export const green = '#61C277';
export const lime = '#53E520';
export const red = '#E52020';

export const successGreen = '#3EB290';

export const pink = '#D8B9C3';
export const darkPurple = '#26204E';
export const balanceWhite = '#F8F8FF';

export const marketDarkGrey = '#3D454B';

export const cardYellow = '#FFEF60';

export const buttonGrey = '#EFEFEF';
