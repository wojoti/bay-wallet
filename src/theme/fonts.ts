export const font = 'Poppins-Light';
export const boldFont = 'Poppins-Bold';
