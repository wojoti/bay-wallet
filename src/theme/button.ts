import {StyleSheet} from 'react-native';
import {primary, secondary} from './colors';
const buttonStyle = StyleSheet.create({
  primary: {
    borderWidth: 0,
    backgroundColor: primary,
  },
  secondary: {
    borderColor: primary,
    backgroundColor: secondary,
  },
});

export const buttonTextStyle = StyleSheet.create({
  primary: {
    color: secondary,
  },
  secondary: {
    color: primary,
  },
});

export default buttonStyle;
