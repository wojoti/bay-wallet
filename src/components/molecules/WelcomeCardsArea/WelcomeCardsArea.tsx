import {Button, Container, Header, Img, ImgType, Label} from 'components/atoms';
import {PropsWithChildren} from 'react';
import {buttonTextStyle} from 'theme/button';
import style from './WelcomeCardsArea.style';

export type WelcomeCardsAreaProps = PropsWithChildren<{
  onButtonPress: () => void;
  mainText: string;
  cardText: string;
  buttonText: string;
  testId?: string;
}>;

const WelcomeCardsArea = ({
  onButtonPress,
  mainText,
  cardText,
  buttonText,
  testId,
}: WelcomeCardsAreaProps) => {
  return (
    <Container flex={1} customStyle={style.wrapper} testId={testId}>
      <Container
        flex={1}
        customStyle={style.cardSizer}
        testId={'welcomecardsarea-card'}>
        <Container
          direction="row"
          justify="space-between"
          customStyle={style.topSpacer}>
          <Container customStyle={style.headerSizer}>
            <Header size={40} testId={'welcomecardsarea-header'}>
              {mainText}
            </Header>
          </Container>
          <Container customStyle={style.logoSizer}>
            <Img
              src={ImgType.logo}
              height={60}
              width={60}
              alt={''}
              testId={'welcomecardsarea-logo'}
            />
          </Container>
        </Container>
        <Container customStyle={style.labelSpacer}>
          <Label size={14} testId={'welcomecardsarea-label'}>
            {cardText}
          </Label>
        </Container>
        <Container align="flex-end" customStyle={style.buttonSpacer}>
          <Button
            customStyle={style.btnStyle}
            customFontStyle={buttonTextStyle.secondary}
            onPress={onButtonPress}
            testId={'welcomecardsarea-button'}>
            {buttonText}
          </Button>
        </Container>
      </Container>
    </Container>
  );
};

export default WelcomeCardsArea;
