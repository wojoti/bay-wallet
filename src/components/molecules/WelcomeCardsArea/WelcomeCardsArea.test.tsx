import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import {primary, secondary} from 'theme/colors';
import WelcomeCardsArea, {WelcomeCardsAreaProps} from './WelcomeCardsArea';

const handlePress = jest.fn();
const props: WelcomeCardsAreaProps = {
  onButtonPress: handlePress,
  mainText: 'maintext',
  cardText: 'cardtext',
  testId: 'test-welcomecardsarea-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<WelcomeCardsArea {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render welcomecardsarea', () => {
  render(<WelcomeCardsArea {...props} />);
  const welcomecardsareaElement = screen.getByTestId(
    'test-welcomecardsarea-id',
  );
  expect(welcomecardsareaElement).toBeOnTheScreen();
});

test('should render welcomecardsarea with header and correct text', () => {
  render(<WelcomeCardsArea {...props} />);
  const welcomecardsareaElement = screen.getByTestId(
    'test-welcomecardsarea-id',
  );
  expect(welcomecardsareaElement).toBeOnTheScreen();
  const welcomecardsareaHeaderElement = within(
    welcomecardsareaElement,
  ).getByTestId('welcomecardsarea-header');
  expect(welcomecardsareaHeaderElement).toBeOnTheScreen();
  const textElement = within(welcomecardsareaHeaderElement).getByText(
    'maintext',
  );
  expect(textElement).toBeOnTheScreen();
});

test('should render welcomecardsarea with label and correct text', () => {
  render(<WelcomeCardsArea {...props} />);
  const welcomecardsareaElement = screen.getByTestId(
    'test-welcomecardsarea-id',
  );
  expect(welcomecardsareaElement).toBeOnTheScreen();
  const welcomecardsareaLabelElement = within(
    welcomecardsareaElement,
  ).getByTestId('welcomecardsarea-label');
  expect(welcomecardsareaLabelElement).toBeOnTheScreen();
  const textElement = within(welcomecardsareaLabelElement).getByText(
    'cardtext',
  );
  expect(textElement).toBeOnTheScreen();
});

test('should render welcomecardsarea with logo', () => {
  render(<WelcomeCardsArea {...props} />);
  const welcomecardsareaElement = screen.getByTestId(
    'test-welcomecardsarea-id',
  );
  expect(welcomecardsareaElement).toBeOnTheScreen();
  const welcomecardsareaImgElement = within(
    welcomecardsareaElement,
  ).getByTestId('welcomecardsarea-logo');
  expect(welcomecardsareaImgElement).toBeOnTheScreen();
});
test('should render welcomecardsarea with button and correct button style', () => {
  render(<WelcomeCardsArea {...props} />);
  const welcomecardsareaElement = screen.getByTestId(
    'test-welcomecardsarea-id',
  );
  expect(welcomecardsareaElement).toBeOnTheScreen();
  const welcomecardsareaButtonElement = within(
    welcomecardsareaElement,
  ).getByTestId('welcomecardsarea-button');
  expect(welcomecardsareaButtonElement).toBeOnTheScreen();
  expect(welcomecardsareaButtonElement).toHaveStyle({
    backgroundColor: secondary,
  });
  const buttonText = within(welcomecardsareaButtonElement).getByTestId(
    'button-text',
  );
  expect(buttonText).toHaveStyle({
    color: primary,
  });
});

test('should handle onButtonPress', () => {
  render(<WelcomeCardsArea {...props} />);
  const welcomecardsareaElement = screen.getByTestId(
    'test-welcomecardsarea-id',
  );
  expect(welcomecardsareaElement).toBeOnTheScreen();
  const welcomecardsareaButtonElement = within(
    welcomecardsareaElement,
  ).getByTestId('welcomecardsarea-button');
  expect(welcomecardsareaButtonElement).toBeOnTheScreen();
  expect(handlePress).toBeCalledTimes(0);
  fireEvent.press(welcomecardsareaButtonElement);
  expect(handlePress).toBeCalledTimes(1);
});
