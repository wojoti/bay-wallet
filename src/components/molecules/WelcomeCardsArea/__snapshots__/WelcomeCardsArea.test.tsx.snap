// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "width": "100%",
      },
    ]
  }
  testID="test-welcomecardsarea-id"
>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "width": "85%",
        },
      ]
    }
    testID="welcomecardsarea-card"
  >
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": undefined,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
          {
            "marginTop": 50,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "stretch",
              "flex": undefined,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "width": "100%",
            },
            {
              "width": "78%",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "black",
              "fontFamily": "Poppins-Bold",
              "fontSize": 40,
              "textAlign": "left",
            }
          }
          testID="welcomecardsarea-header"
        >
          maintext
        </Text>
      </View>
      <View
        style={
          [
            {
              "alignItems": "stretch",
              "flex": undefined,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "width": "100%",
            },
            {
              "width": "auto",
            },
          ]
        }
      >
        <SvgMock
          height={60}
          testID="welcomecardsarea-logo"
          width={60}
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          {
            "marginTop": 20,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontFamily": "Poppins-Light",
            "fontSize": 14,
            "fontWeight": "400",
            "textAlign": "left",
          }
        }
        testID="welcomecardsarea-label"
      >
        cardtext
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "flex-end",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          {
            "marginTop": 30,
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 65,
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "borderWidth": 0,
            },
          ]
        }
        testID="welcomecardsarea-button"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Bold",
                "fontSize": 18,
                "textAlign": "center",
              },
              {
                "color": "#363062",
              },
            ]
          }
          testID="button-text"
        >
          buttontext
        </Text>
      </View>
    </View>
  </View>
</View>
`;
